GTG/gtk/browser/main_window.py

```diff
@@ -1282,45 +1282,50 @@ def on_dismiss_task(self, widget=None):
                 task.set_status(Task.STA_DISMISSED)
                 self.close_all_task_editors(uid)
 
-    def apply_filter_on_panes(self, filter_name, refresh=True, parameters=None):
-        """ Apply filters for every pane: active tasks, closed tasks """
-
-        for pane in self.vtree_panes:
-            vtree = self.req.get_tasks_tree(name=pane, refresh=False)
-            vtree.apply_filter(filter_name, refresh=refresh, parameters=parameters)
-
-    def unapply_filter_on_panes(self, filter_name, refresh=True):
-        """ Apply filters for every pane: active tasks, closed tasks """
-        for pane in self.vtree_panes:
-            vtree = self.req.get_tasks_tree(name=pane, refresh=False)
-            vtree.unapply_filter(filter_name, refresh=refresh)
+    def _reapply_filter(self, current_pane):
+        filters = self.get_selected_tags()
+        filters.append(current_pane)
+        vtree = self.req.get_tasks_tree(name=current_pane, refresh=False)
+        # Re-applying search if some search is specified
+        search = self.search_entry.get_text()
+        if search:
+            filters.append(SEARCH_TAG)
+        # only reseting filters if applieds filters are different from current
+        # ones, leaving a chance for liblarch to make the good call on wether
+        # to refilter or not
+        if sorted(filters) != sorted(vtree.list_applied_filters()):
+            vtree.reset_filters(refresh=False)
+        # browsing and applying filters
+        for filter_ in filters:
+            is_last = filter_ == filters[-1]
+            if filter_ == SEARCH_TAG:
+                self._try_filter_by_query(search, refresh=is_last)
+            else:
+                vtree.apply_filter(filter_, refresh=is_last)
```

Comment:
Well, `filter` is a python reserved keyword, and it's not recommended as a rule to override reserved names, as for `filter_` especially, it's the [PEP8 recommended behavior.](https://legacy.python.org/dev/peps/pep-0008/#function-and-method-arguments). But I can find a synonym. 

I'm not a big fan of the `is_last` thingy either, and I'd rather prefer explicitly call for `refilter()`. Better yet, a new `liblarch` method allowing to apply multiple filters at once. But since I didn't want to alter too much of the `liblarch` API and still wanted to limit calls I thought of this. But yeat, a comment is indeed needed.

Reply:
> Well, filter is a python reserved keyword, and it's not recommended as a rule to override reserved names, as for filter_ especially, it's the PEP8 recommended behavior.. But I can find a synonym.

But `filter` is a [built-in function](https://docs.python.org/3/library/functions.html#filter) (or class as the python3 interpreter says), not a reserved [keyword](https://docs.python.org/3/reference/lexical_analysis.html#keywords) from what I can see. I can redefine it at least in the python REPL, so I'm sure it isn't a keyword.

Reply:
Yes indeed, keyword and not built-in function, sorry about the mix up. 
But to me it's still overriding a declared value. From experience it may lead to some complicated debugging. So yeah I usually follow the PEP8 recommendation about keywords on that too. 
But since it's important I'll use the name from the `ViewTree.apply_filter` function.